libs/utils/trace.py

```diff
@@ -169,16 +163,22 @@ def setXTimeRange(self, t_min=None, t_max=None):
         :param t_max: upper bound
         :type t_max: int or float
         """
-        if t_min is None:
-            self.x_min = 0
-        else:
+        self.x_min = self.trace.window[0]
+        self.x_max = self.trace.window[1]
+        if t_min is not None:
+            if t_min < self.x_min or t_min > self.x_max:
+                raise ValueError('t_min out of range: '
```

Comment:
Maybe in the first case we can just dump a warning and cap the value to the trace minimum?

```python
if t_min < self.x_min:
        logging.warning('t_min out of range: capping to trace minimum %.6f', self.x_min)
if t_min > self.x_max:
        raise ValueError('t_min out of range: trace boundaries are ({:.6f}, {:.6f}) [s]',
        .format(self.x_min, self.x_max))
```